elixir/apps/domain/lib/domain/config/definitions.ex

```diff
@@ -584,8 +584,15 @@ defmodule Domain.Config.Definitions do
     default: %{},
     sensitive: true,
     dump: fn map ->
-      Dumper.keyword(map)
-      |> Keyword.update(:tls_options, nil, &Dumper.dump_ssl_opts/1)
+      keyword = Dumper.keyword(map)
+        |> Keyword.update(:tls_options, nil, &Dumper.dump_ssl_opts/1)
+
+      cond do
+        keyword[:ssl] ->
+          Keyword.update(keyword, :sockopts, nil, &Dumper.dump_ssl_opts/1)
+        true ->
+          keyword
+      end
```

Comment:
Maybe we can always map that key to make the code simpler?
```suggestion
      Dumper.keyword(map)
      |> Keyword.update(:tls_options, nil, &Dumper.dump_ssl_opts/1)
      |> Keyword.update(:sockopts, nil, &Dumper.dump_ssl_opts/1)
```

Reply:
Let's do it like this for now. This `cond` was useless anyway because when ssl is disabled `sockopts` probably needs a special dumper (like it is done with ssl_opts)